File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/data/mapper/SortOrderEntityMapper.java
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.data.mapper;



import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.data.model.SortOrderEntity;
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.model.SortOrder;

import javax.inject.Inject;

/**
 * Created by MateuszW on 2018-05-27.
 */
public class SortOrderEntityMapper {

    @Inject
    public SortOrderEntityMapper() {
    }

    public SortOrder entityToDomain(SortOrderEntity sortOrderEntity) {
        SortOrder sortOrder = null;
        switch (sortOrderEntity) {
            case TITLE:
                return SortOrder.TITLE;
            case NO_OF_EPISODES:
                return SortOrder.NO_OF_EPISODES;
        }
        return sortOrder;
    }

    public SortOrderEntity domainToEntity(SortOrder sortOrder) {
        SortOrderEntity sortOrderEntity = null;
        switch (sortOrder) {
            case TITLE:
                return SortOrderEntity.TITLE;
            case NO_OF_EPISODES:
                return SortOrderEntity.NO_OF_EPISODES;
        }
        return sortOrderEntity;
    }
}

File: app/src/main/java/com/wolkowiczmateusz/android/cleanmvpretrofitrxdaggerboilerplate/data/mapper/SortOrderEntityMapper.kt
package com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.data.mapper

import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.data.model.SortOrderEntity
import com.wolkowiczmateusz.android.cleanmvpretrofitrxdaggerboilerplate.domain.model.SortOrder

import javax.inject.Inject

class SortOrderEntityMapper @Inject
constructor() {

    fun entityToDomain(sortOrderEntity: SortOrderEntity): SortOrder? {
        val sortOrder: SortOrder? = null
        return when (sortOrderEntity) {
            SortOrderEntity.TITLE -> SortOrder.TITLE
            SortOrderEntity.NO_OF_EPISODES -> SortOrder.NO_OF_EPISODES
        }
        return sortOrder
    }

    fun domainToEntity(sortOrder: SortOrder): SortOrderEntity? {
        val sortOrderEntity: SortOrderEntity? = null
        return when (sortOrder) {
            SortOrder.TITLE -> SortOrderEntity.TITLE
            SortOrder.NO_OF_EPISODES -> SortOrderEntity.NO_OF_EPISODES
        }
        return sortOrderEntity
    }
}
